### PR TITLE
openstack/clientconfig/requests.go: make sure tls config is generated before creating authenticated client

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Previously the openstack.AuthenticatedClient call would create provider and use it to create an authenticated client without the required TLS configuration.

Now the function creates a provider client and updates the HTTPClient before calling Authenticate to setup the client with authentication tokens.

### CODE REVIEW

File: openstack/clientconfig/requests.go [hunk 1]
```diff
@@ -745,7 +745,11 @@ func NewServiceClient(service string, opts *ClientOpts) (*gophercloud.ServiceCli
 	}
 
 	// Get a Provider Client
-	pClient, err := AuthenticatedClient(opts)
+	ao, err := AuthOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+	pClient, err := openstack.NewClient(ao.IdentityEndpoint)
```
Comment (mandre) [inline openstack/clientconfig/requests.go:752]: How about we take the opportunity to move the creation of `pClient` closer to where it's used, on line 803?

**Review by Fedosin — APPROVED** [review]
lgtm

File: openstack/clientconfig/requests.go (hunk 1 above)
Comment (abhinavdahiya) [reply openstack/clientconfig/requests.go]: @mandre fixed with 9cc2ea9

**Review by mandre — COMMENTED** [review]
Thanks @abhinavdahiya . LGTM :+1:

**Review by jtopjian — APPROVED** [review]
LGTM

Thanks for catching this! Sorry about that.

**Comment by jtopjian** [comment]
Thanks for the reviews, everyone.

It's been almost 4 years since I first started working with TLS and OpenStack authentication. I recall that there were handful of edge cases that were reported, but once it was working with Terraform, I took it for granted and set it aside in my head. From reviewing this implementation with Terraform's, I think this should be good to go, but my memory is a little fuzzy on those edge cases. 

If any are identified, just open a PR and I'm happy to merge.

**Pull request merged**